### PR TITLE
Use GroupInfo instead of PublicGroupState. (1/2)

### PR DESCRIPTION
**Pull request by duesee**
This PR introduces the `external_pub` extension needed to make progress on #719 and #1027. This is basically the (unchanged) start of closed #1106.

### CODE REVIEW

File: openmls/src/extensions/external_pub_extension.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+/// ```c
+/// // draft-ietf-mls-protocol-16
+/// struct {
+///     HPKEPublicKey external_pub;
+/// } ExternalPub;
+/// ```
+#[derive(
+    PartialEq, Eq, Clone, Debug, Serialize, Deserialize, TlsSerialize, TlsDeserialize, TlsSize,
+)]
+pub struct ExternalPubExtension {
```
Comment (franziskuskiefer) [inline openmls/src/extensions/external_pub_extension.rs:15]: Wasn't there a test for this extension? Would be nice to have at least something along the lines of the others in `test_extensions`.

File: openmls/src/extensions/mod.rs [hunk 2]
```diff
@@ -85,30 +88,37 @@ pub enum ExtensionType {
...
+    /// tree.
+    RatchetTree = 2,
+
+    /// The required capabilities extension defines the configuration of a group
+    /// that imposes certain requirements on clients in the group.
+    RequiredCapabilities = 3,
+
+    /// To join a group via an External Commit, a new member needs a GroupInfo
+    /// with an ExternalPub extension present in its extensions field.
+    ExternalPub = 4,
```
Comment (franziskuskiefer) [inline openmls/src/extensions/mod.rs:105]: Can you add external sender while you're at it? Then we have all values from the RFC.

File: openmls/src/extensions/mod.rs [hunk 3]
```diff
@@ -85,30 +88,37 @@ pub enum ExtensionType {
     /// Reserved. This must not be used.
     Reserved = 0,
```
Comment (franziskuskiefer) [inline openmls/src/extensions/mod.rs:89]: Can you drop this one? It should not be able to instantiate and OpenMLS moved towards not defining the reserved value.

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Thanks for splitting this out. Just a few nits.

File: openmls/src/extensions/external_pub_extension.rs (hunk 1 above)
Comment (duesee) [reply openmls/src/extensions/external_pub_extension.rs:15]: No there wasn't. But I will add something :+1:

Comment (duesee) [reply openmls/src/extensions/external_pub_extension.rs:15]: Ah, I think you thought about this one https://github.com/openmls/openmls/pull/1085/files#diff-7f947c709686d3816215e94503d38aafeb8990e9a8b60230fedbbc1aed50bfd8R31 I was also confused for a moment :-)

**Review by franziskuskiefer — APPROVED** [review]
Thanks!

**Review by franziskuskiefer — APPROVED** [review]

**Pull request merged**